code/ship/ship.cpp

```diff
@@ -7450,19 +7452,24 @@ void ship_render_player_ship(object* objp) {
 
 	const bool hasCockpitModel = sip->cockpit_model_num >= 0;
 
-	const bool renderCockpitModel = Viewer_mode != VM_TOPDOWN && hasCockpitModel;
+	const bool renderCockpitModel = (Viewer_mode != VM_TOPDOWN) && hasCockpitModel && !disableCockpits;
 	const bool renderShipModel = (sip->flags[Ship::Info_Flags::Show_ship_model])
 		&& (!Viewer_mode || (Viewer_mode & VM_PADLOCK_ANY) || (Viewer_mode & VM_OTHER_SHIP) || (Viewer_mode & VM_TRACK)
 			|| !(Viewer_mode & VM_EXTERNAL));
 
 	//Nothing to do
-	if (!(renderCockpitModel || renderShipModel))
+	if (!(renderCockpitModel || renderShipModel)) {
+		cockpitActive = false;
```

Comment:
cockpitActive should not be set if we render cockpit or show ship, but likely to the value of ``renderCockpitModel`` directly, which corresponds to whether a cockpit is drawn or not, as opposed to also depending on the show ship flag.